services/catalog/src/simcore_service_catalog/core/background_tasks.py

```diff
@@ -117,59 +130,105 @@ async def _get_service_access_rights(
+        (service.key, service.version)
+        for service in await services_repo.list_services(
+            gids=[everyone_gid], execute_access=True
+        )
+    }
+
+    missing_services = published_services - available_services
+    missing_services_access_rights = [
+        ServiceAccessRightsAtDB(
+            key=service[0], version=service[1], gid=everyone_gid, execute_rights=True
+        )
+        for service in missing_services
+    ]
+    if missing_services_access_rights:
+        logger.info(
+            "Adding access rights for published templates\n: %s",
+            missing_services_access_rights,
+        )
+        await services_repo.upsert_service_access_rights(missing_services_access_rights)
 
 
 async def sync_registry_task(app: FastAPI) -> None:
     # get list of services from director
-    while True:
-        try:
-            logger.debug("syncing services between registry and database...")
-            services_in_registry: Dict[
-                Tuple[ServiceKey, ServiceVersion], ServiceDockerData
-            ] = await _list_registry_services(app)
-            services_in_db: Set[
-                Tuple[ServiceKey, ServiceVersion]
-            ] = await _list_db_services(app)
-
-            # check that the db has all the services at least once
-            missing_services_in_db = set(services_in_registry.keys()) - services_in_db
-            if missing_services_in_db:
-                logger.debug(
-                    "missing services in db:\n%s", pformat(missing_services_in_db),
-                )
-                # update db (rationale: missing services are shared with everyone for now)
-                await _create_services_in_db(
-                    app, missing_services_in_db, services_in_registry
-                )
-
-            await asyncio.sleep(app.state.settings.background_task_rest_time)
-
-        except CancelledError:
-            # task is stopped
-            return
-        except Exception:  # pylint: disable=broad-except
-            logger.exception("Error while processing services entry")
-            await asyncio.sleep(
-                5
-            )  # wait a bit before retrying, so it does not block everything until the director is up
+    engine: Engine = app.state.engine
+    async with engine.acquire() as conn:
+        while True:
+            try:
+                logger.debug("syncing services between registry and database...")
+                # check that the list of services is in sync with the registry
+                await _ensure_services_metadata_uptodate(app, conn)
+
+                # check that the published services are available to everyone (templates are published to GUESTs, so their services must be also accessible)
+                await _ensure_published_services_are_accessible(conn)
+
+                await asyncio.sleep(app.state.settings.background_task_rest_time)
+
+            except CancelledError:
+                # task is stopped
```

Comment:
log debug the reason fo the cancelation?

Reply:
I can add. there is only one possible cancelation in that case, which is when the app stops the background task.